resolver.go

```diff
@@ -36,12 +36,18 @@ const (
 	ptrIPv4domain = ".in-addr.arpa."
 	ptrIPv6domain = ".ip6.arpa."
 	respTTL       = 1800
+	maxExtDNS     = 3 //max number of external servers to try
 )
 
+type extDNSEntry struct {
```

Comment:
Makes sense, also I misread the code, you actually keep the `IP` as string, so never mind.
Got confused by the IP parsing. I, in fact, would just look for a signature since it seems the addresses are already validated, something like:
```
r.extDNS[i].ip = dns[i]
r.extDNS[i].isIPv6 = strings.Contains(dns[i], ":")
```
but this is minor, up to you.